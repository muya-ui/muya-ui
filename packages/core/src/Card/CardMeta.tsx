import React from 'react';

import { Typography } from '../Typography';
import useTheme from '../utils/useTheme';
import { StyleCardMeta, StyleCardTitle } from './styled';
import { ICardMetaProps } from './types';
import memoForwardRef from '../utils/memoForwardRef';

const CardMeta = memoForwardRef<HTMLDivElement, ICardMetaProps>((props, ref) => {
  const { children, title, text, space = 4, ...others } = props;
  const theme = useTheme();
  return (
    <StyleCardMeta ref={ref} {...others}>
      {title ? (
        <StyleCardTitle level={theme.components.Card.defaultMetaTitleLevel} $space={space}>
          {title}
        </StyleCardTitle>
      ) : null}
      {text ? <Typography.Text>{text}</Typography.Text> : null}
      {children}
    </StyleCardMeta>
  );
});

export default CardMeta;
