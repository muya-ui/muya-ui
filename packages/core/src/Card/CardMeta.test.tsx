import React from 'react';
import renderer from 'react-test-renderer';
import CardMeta from './CardMeta';

describe('CardMeta', () => {
  it('should render success', function() {
    const wrapper = renderer
      .create(
        <>
          <CardMeta>测试</CardMeta>
          <CardMeta title="测试" />
          <CardMeta title={<span>测试</span>} />
          <CardMeta text="测试" />
          <CardMeta text={<span>测试</span>} />
          <CardMeta title="测试" text="测试" />
          <CardMeta title={<span>测试</span>} text={<span>测试</span>} />
          <CardMeta title="测试" text="测试">
            测试
          </CardMeta>
          <CardMeta title={<span>测试</span>} text={<span>测试</span>}>
            测试
          </CardMeta>
        </>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
