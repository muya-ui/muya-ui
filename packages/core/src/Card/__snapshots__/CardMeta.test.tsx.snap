// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMeta should render success 1`] = `
Array [
  <div
    className=""
  >
    测试
  </div>,
  .c1 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  margin-bottom: 0;
}

.c1 + .c0 {
  margin-top: 0;
}

.c2 {
  margin-bottom: 4px;
}

<div
    className=""
  >
    <h4
      className="c0 c1 c2"
    >
      测试
    </h4>
  </div>,
  .c1 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  margin-bottom: 0;
}

.c1 + .c0 {
  margin-top: 0;
}

.c2 {
  margin-bottom: 4px;
}

<div
    className=""
  >
    <h4
      className="c0 c1 c2"
    >
      <span>
        测试
      </span>
    </h4>
  </div>,
  .c0 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

<div
    className=""
  >
    <span
      className="c0"
    >
      测试
    </span>
  </div>,
  .c0 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

<div
    className=""
  >
    <span
      className="c0"
    >
      <span>
        测试
      </span>
    </span>
  </div>,
  .c3 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  margin-bottom: 0;
}

.c1 + .c0 {
  margin-top: 0;
}

.c2 {
  margin-bottom: 4px;
}

<div
    className=""
  >
    <h4
      className="c0 c1 c2"
    >
      测试
    </h4>
    <span
      className="c3"
    >
      测试
    </span>
  </div>,
  .c3 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  margin-bottom: 0;
}

.c1 + .c0 {
  margin-top: 0;
}

.c2 {
  margin-bottom: 4px;
}

<div
    className=""
  >
    <h4
      className="c0 c1 c2"
    >
      <span>
        测试
      </span>
    </h4>
    <span
      className="c3"
    >
      <span>
        测试
      </span>
    </span>
  </div>,
  .c3 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  margin-bottom: 0;
}

.c1 + .c0 {
  margin-top: 0;
}

.c2 {
  margin-bottom: 4px;
}

<div
    className=""
  >
    <h4
      className="c0 c1 c2"
    >
      测试
    </h4>
    <span
      className="c3"
    >
      测试
    </span>
    测试
  </div>,
  .c3 {
  color: rgba(102,102,102,1);
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  margin: 0;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  margin-bottom: 0;
}

.c1 + .c0 {
  margin-top: 0;
}

.c2 {
  margin-bottom: 4px;
}

<div
    className=""
  >
    <h4
      className="c0 c1 c2"
    >
      <span>
        测试
      </span>
    </h4>
    <span
      className="c3"
    >
      <span>
        测试
      </span>
    </span>
    测试
  </div>,
]
`;
